src/interface/truncation.jl

```diff
@@ -42,85 +42,102 @@ Truncation strategy that does nothing, and keeps all the values.
 notrunc() = NoTruncation()
 
 """
-    TruncationKeepSorted(howmany::Int, by::Function, rev::Bool)
+    TruncationByOrder(howmany::Int, by::Function, rev::Bool)
 
 Truncation strategy to keep the first `howmany` values when sorted according to `by` in increasing (decreasing) order if `rev` is false (true).
+
 See also [`truncrank`](@ref).
 """
-struct TruncationKeepSorted{F} <: TruncationStrategy
+struct TruncationByOrder{F} <: TruncationStrategy
     howmany::Int
     by::F
     rev::Bool
 end
 
 """
-    truncrank(howmany::Int; by=abs, rev=true)
+    truncrank(howmany::Integer; by=abs, rev::Bool=true)
 
 Truncation strategy to keep the first `howmany` values when sorted according to `by` or the last `howmany` if `rev` is true.
 """
-truncrank(howmany::Int; by=abs, rev=true) = TruncationKeepSorted(howmany, by, rev)
+truncrank(howmany::Integer; by=abs, rev::Bool=true) = TruncationByOrder(howmany, by, rev)
 
 """
-    TruncationKeepFiltered(filter::Function)
+    TruncationByFilter(filter::Function)
 
 Truncation strategy to keep the values for which `filter` returns true.
+
+See also [`truncfilter`](@ref).
 """
-struct TruncationKeepFiltered{F} <: TruncationStrategy
+struct TruncationByFilter{F} <: TruncationStrategy
     filter::F
 end
 
 """
-    trunctol(val::Real; by=abs)
+    truncfilter(filter)
 
-Truncation strategy to discard the values that are smaller than `val` according to `by`.
+Truncation strategy to keep the values for which `filter` returns true.
 """
-trunctol(val::Real; by=abs) = TruncationKeepFiltered(≥(val) ∘ by)
+truncfilter(f) = TruncationByFilter(f)
 
 """
-    truncabove(val::Real; by=abs)
+    TruncationByValue(atol::Real, rtol::Real, p::Real, by, rev::Bool=true)
 
-Truncation strategy to discard the values that are larger than `val` according to `by`.
+Truncation strategy to keep the values that satisfy `by(val) < max(atol, rtol * norm(values, p)`
```

Comment:
Is the default to keep smaller values? Is the default value `rev = true`? I would think we want `rev=false` as default, and then for the docstring
```markdown
Truncation strategy to keep the values that satisfy
`by(val) >= max(atol, rtol * norm(values, p)`, or
to discard those values if `rev = true`.
```